spec.html

```diff
@@ -36755,6 +36755,17 @@ <h1>Structured Data</h1>
   <emu-clause id="sec-arraybuffer-objects">
     <h1>ArrayBuffer Objects</h1>
 
+    <emu-clause id="sec-arraybuffer-notation">
+      <h1>Notation</h1>
+      <p>The descriptions below in this section, <emu-xref href="#sec-atomics-object"></emu-xref>, and <emu-xref href="#sec-memory-model"></emu-xref> use the read-modify-write modification function internal data structure.</p>
+      <p>A <em>read-modify-write modification function</em> is a mathematical function that is notationally represented as an abstract closure that takes two Lists of bytes as arguments returns a List of bytes. These abstract closures satisfy all of the following properties:</p>
+      <ul>
+        <li>They do not access any language or specification values except their parameters, captured values, and the [[LittleEndian]] field of the surrounding agent's Agent Record.</li>
+        <li>They do not return completion values and do not invoke other abstract closures or operations that return completion values.</li>
```

Comment:
Shouldn't "they do not return completion values" be deferred until we do overall completion value reform?

Reply:
Well, I think it's important they're obviously "pure" from the start, and returning explicit Completion Records would confuse that. Calling out something like "do not return explicit completion values and do not invoke other [...] that return explicit completion values" is also kind of confusing IMO. Any ideas here?

Reply:
I don't see how that would confuse it at all - if you wanted to constrain it to only return *normal* completion values, that seems fine, but with the current state of completion values, having some of them *not* return one seems very confusing to me.